doc/specs/#1502 - Advanced Tab Switcher/spec.md

```diff
@@ -0,0 +1,219 @@
+2. <kbd>esc</kbd> : dismisses the UI without changing tab focus.
+
+The following are ways a user can dismiss the UI, _whether or not_ the `Anchor` arg is provided to `OpenTabSwitcher`.
+
+1. The user can press a number associated with a tab to instantly switch to the tab and dismiss the switcher.
+2. The user can click on a tab to instantly switch to the tab and dismiss the switcher.
+3. The user can click outside of the UI to dismiss the switcher without bringing the selected tab into focus.
+4. The user can press any of the dismissal keybinds.
+
+If the `Anchor` arg is provided, then in addition to the above methods, the UI will dismiss upon the release of the `Anchor` key.
+
+Pressing the `OpenTabSwitcher` keychord again will _not_ close the Switcher.
+
+### Most Recently Used Order
+
+We'll provide a setting that will allow the list of tabs to be presented in either _in-order_ (how the tabs are ordered on the tab bar), or _Most Recently Used Order_ (MRU). MRU means that the tab that the terminal most recently visited will be on the top of the list, and the tab that the terminal has not visited for the longest time will be on the bottom.
+
+### Numbered Tabs
+
+Similar to how the user can currently switch to a particular tab with a combination of keys such as <kbd>ctrl+shift+1</kbd>, we want to have the tab switcher provide a number to the first nine tabs (1-9) in the list for quick switching. If there are more than nine tabs in the list, then the rest of the tabs will not have a number assigned.
+
+## Capabilities
+
+### Accessibility
+
+- The tab switcher will be using WinUI, and so it'll be automatically linked to the UIA tree. This allows screen readers to find it, and so narrator will be able to navigate the switcher easily.
+- The UI is also fully keyboard-driven, with the option of using a mouse to interact with the UI.
+- When the tab switcher pops up, the focus immediately swaps to it.
+- For the sake of contrast with the background, we could make it so that it's similar to how the About Page is presented.
+Specifically, a box would pop up, and the rest of the terminal would turn white with some transparency so that the focus is very clearly on the tab switcher.
+
+### Security
+
+This shouldn't introduce any security issues.
+
+### Reliability
+
+How we're updating the MRU is something to watch out for since it triggers on a lot of tab interactions. However, I don't foresee the update taking long at all, and I can't imagine that users can create and delete tabs fast enough to matter.
+
+### Compatibility
+
+- The existing way of navigating horizontally through the tabs on the tab bar should not break.
+- These should also be separate keybindings from the keybindings associated with using the tab switcher.
+- Reordering tabs on the tab bar shouldn't change the MRU order. The MRU is essentially supposed to be a different _view_ of the already existing tab list.

```

Comment:
Wait hold up - if I have tabs that are:
```[cmd(focused), ps, wsl]```
in MRU order: cmd, ps, wsl, (0, 1, 2) then reorder to 
```[wsl, cmd(focused), ps]```
the MRU order remains unchanged, right? It's not changed to wsl, cmd, ps to match the previous (0, 1, 2) ordering, right? I think I got confused reading this...

Reply:
You're totally right, sorry for the confusion, I updated this point and even used the example you gave. 👍